pkg/support/collectors.go

```diff
@@ -102,3 +113,43 @@ func (c *collectorFactory) DefaultCollectors() []*Collect {
 		},
 	}
 }
+
+func (c *collectorFactory) EksaHostCollectors(osFamilies map[v1alpha1.OSFamily]bool) []*Collect {
+	for family := range osFamilies {
+		switch family {
+		case v1alpha1.Ubuntu:
+			return c.ubuntuHostCollectors()
```

Comment:
Instead of making it a switch here, can we just have a map of osFamily to the appropriate method? That way instead of having to loop through all the machine config objects at the top to create our set and then also setting it to true, we can just retrieve the appropriate collectors based on the machine config. Maybe I am missing some other reason as to why we are doing it this way though.

Reply:
ya makes sense

Reply:
this is intended to easily support heterogeneous OS families in the future. I'm using a 'set' so we 

- only end up with a single entry per os family, make it easy to avoid generating redundant configs. 
- allow the 'collectorFactory' to determine what collectors we generate, making it easier to determine shared collectors, group collectors, etc...this is going to increase in complexity once we add CP vs WN specific collectors, heterogeneous clusters with both BR & Ubuntu, etc, so I think keeping all the logic in the factory will be good

that being said, we can get rid of the switch and just check for the presence of the specific OS family in the map with 'ok' or w/e, that's fine by me

Reply:
ok updated